src/utils/PolygonConnector.h

```diff
@@ -179,7 +586,154 @@ class PolygonConnector
+                if(connection_length < max_gap * line_width) //Connection is allowed.
+                {
+                    result = connection;
+                    best_connection_length = connection_length;
+                }
+            }
+            if(to_backward_intersection)
+            {
+                PolygonConnection<Polygonal> connection(first.from_poly, from_forward_intersection->second, from_forward_intersection->first, first.to_poly, to_backward_intersection->second, to_backward_intersection->first);
+                const coord_t connection_length = getSpace(connection);
+                if(connection_length < max_gap * line_width && connection_length < best_connection_length) //Better than the previous best.
+                {
+                    result = connection;
+                    best_connection_length = connection_length;
+                }
+            }
+        }
+        if(from_backward_intersection)
+        {
+            //Find the shortest of the connections in the to_poly.
+            if(to_forward_intersection)
+            {
+                PolygonConnection<Polygonal> connection(first.from_poly, from_backward_intersection->second, from_backward_intersection->first, first.to_poly, to_forward_intersection->second, to_forward_intersection->first);
+                const coord_t connection_length = getSpace(connection);
+                if(connection_length < max_gap * line_width) //Connection is allowed.
+                {
+                    result = connection;
+                    best_connection_length = connection_length;
+                }
+            }
+            if(to_backward_intersection)
+            {
+                PolygonConnection<Polygonal> connection(first.from_poly, from_backward_intersection->second, from_backward_intersection->first, first.to_poly, to_backward_intersection->second, to_backward_intersection->first);
+                const coord_t connection_length = getSpace(connection);
+                if(connection_length < max_gap * line_width && connection_length < best_connection_length) //Better than the previous best.
+                {
+                    result = connection;
+                    best_connection_length = connection_length;
+                }
+            }
+        }
+        return result;
+    }
+
+    template<typename Polygonal>
+    void connectPolygonsAlongBridge(const PolygonBridge<Polygonal>& bridge, Polygonal& result)
+    {
+        //We'll traverse the following path:
+        //
+        // <<<<<<X......X<<<<<<< to_poly
+        //       ^      v
+        //       ^      v
+        //       ^ a  b v bridge
+        //       ^      v
+        // >>>>>>X......X>>>>>>> from_poly
+        //
+        //To do this, from_poly and to_poly might need to be traversed in reverse order. This function figures all of that out.
+
+        Polygonal ret; //Create a temporary result that we'll move into the result.
```

Comment:
When called from `Infill::generate` ->`PolygonConnector::connect` ->`connectGroup` ->`connectPolygonsAlongBridge`, where `Polygon` is instantiated to `ExtrusionLine`, the default constructor writes an `inset_idx = SIZE_MAX`. 
This latter causes an invalid access in [`InsetOrderOptimizer::getInsetOrder`](https://github.com/Ultimaker/CuraEngine/blob/dec7a07ca07b78a8884ef5f4cfc633856ecccf44/src/InsetOrderOptimizer.cpp#L326) after the `fillers_by_inset` vector is resized to `SIZE_MAX+1=0`.

Reply:
Sounds like a problem indeed. I'll look into it.

Do you have a test case that reproduces this issue? Or does it happen always and just doesn't crash on it?

Reply:
I get a segmentation fault (Debug & release builds on linux). All I had to do is to enable concentric Top/Bottom and Connect Top/Bottom Polygons, on top of FFF's Normal profile.

[WrappingInsetIdx.zip](https://github.com/Ultimaker/CuraEngine/files/8253455/WrappingInsetIdx.zip)

Reply:
I found a case where this seems to be the problem: [crash_on_order_by_inset.zip](https://github.com/Ultimaker/CuraEngine/files/8253683/crash_on_order_by_inset.zip)

I have a fix for it. Thank you for the report!